src/App.js

```diff
@@ -1,64 +1,77 @@
 import React, { Component } from "react";
 import "./App.css";
 
-import Card from "./components/Card/Card";
 import AddCard from "./components/AddCard/AddCard";
+import AllCards from "./components/AllCards/AllCards";
+import Search from "./components/Search/Search";
 
 class App extends Component {
-  state = {
-    cards: [
-      {
-        id: 1,
-        name: "John",
-        email: "john@gmail.com"
-      },
-      {
-        id: 2,
-        name: "Rob",
-        email: "rob@gmail.com"
-      }
-    ],
-    showAddCard: false,
-    lastId: 2
-  };
+  constructor() {
+    super();
+    this.state = {
+      cards: [],
+      showAddCard: false,
+      lastId: 2,
+      filterText: ""
+    };
+  }
+
+  componentWillMount() {
+    this.setState({
+      cards: [
+        {
+          id: 1,
+          name: "John",
+          email: "john@gmail.com"
+        },
+        {
+          id: 2,
+          name: "Rob",
+          email: "rob@gmail.com"
+        }
+      ]
+    });
+  }
 
   onShowAddCard = () => {
     const { showAddCard } = this.state;
     this.setState({ showAddCard: !showAddCard });
   };
 
-  onAddCard = (card) => {
+  onAddCard = card => {
     const { cards, lastId } = this.state;
-    cards.push({...card,id: lastId+1}); //TODO find another way to generate id
-    console.log(card);
-
-    this.setState({cards: cards, lastId: lastId+1});
-  }
+    cards.unshift({ ...card, id: lastId + 1 });
+    this.setState({ cards: cards, lastId: lastId + 1 });
+  };
 
-  onDeleteCard = (id) => {
+  onDeleteCard = id => {
     const { cards } = this.state;
-    const index = cards.findIndex((card) => {
+    const index = cards.findIndex(card => {
```

Comment:
With fat arrow we can use shoarthand notation in situation when it immediately returns result  `card => card.id === id`